GH-41964: WIP: [CI][C++] Debug mamba error

kou opened the pull request:
<!--
Thanks for opening a pull request!
If this is your first pull request you can find detailed information on how 
to contribute here:
  * [New Contributor's Guide](https://arrow.apache.org/docs/dev/developers/guide/step_by_step/pr_lifecycle.html#reviews-and-merge-of-the-pull-request)
  * [Contributing Overview](https://arrow.apache.org/docs/dev/developers/overview.html)


If this is not a [minor PR](https://github.com/apache/arrow/blob/main/CONTRIBUTING.md#Minor-Fixes). Could you open an issue for this pull request on GitHub? https://github.com/apache/arrow/issues/new/choose

Opening GitHub issues ahead of time contributes to the [Openness](http://theapacheway.com/open/#:~:text=Openness%20allows%20new%20users%20the,must%20happen%20in%20the%20open.) of the Apache Arrow project.

Then could you also rename the pull request title in the following format?

    GH-${GITHUB_ISSUE_ID}: [${COMPONENT}] ${SUMMARY}

or

    MINOR: [${COMPONENT}] ${SUMMARY}

In the case of PARQUET issues on JIRA the title also supports:

    PARQUET-${JIRA_ISSUE_ID}: [${COMPONENT}] ${SUMMARY}

-->

### Rationale for this change

<!--
 Why are you proposing this change? If this is already explained clearly in the issue then this section is not needed.
 Explaining clearly why changes are proposed helps reviewers understand your changes and offer better suggestions for fixes.  
-->

### What changes are included in this PR?

<!--
There is no need to duplicate the description in the issue here but it is sometimes worth providing a summary of the individual changes in this PR.
-->

### Are these changes tested?

<!--
We typically require tests for all PRs in order to:
1. Prevent the code from being accidentally broken by subsequent changes
2. Serve as another way to document the expected behavior of the code

If tests are not included in your PR, please explain why (for example, are they covered by existing tests)?
-->

### Are there any user-facing changes?

<!--
If there are user-facing changes then we may require documentation to be updated before approving the PR.
-->

<!--
If there are any breaking changes to public APIs, please uncomment the line below and explain which changes are breaking.
-->
<!-- **This PR includes breaking changes to public APIs.** -->

<!--
Please uncomment the line below (and provide explanation) if the changes fix either (a) a security vulnerability, (b) a bug that caused incorrect or invalid data to be produced, or (c) a bug that causes a crash (even when the API contract is upheld). We use this to highlight fixes to issues that may affect users without their knowledge. For this reason, fixing bugs that cause errors don't count, since those are usually obvious.
-->
<!-- **This PR contains a "Critical Fix".** -->

pitrou reviewed (approved):
+1, thanks a lot @kou

kou commented:
> Is this something that should be reported to the mamba issue tracker?

Hmm... Maybe no...?
It seems that this is related to our AppVeyor environment.

This is the last succeeded job before this problem is happen. Note that mamba 1.5.8 was selected:

https://ci.appveyor.com/project/ApacheSoftwareFoundation/arrow/builds/49866639?fullLog=true#L90

```text
C:\projects\arrow>conda install -q -y -c conda-forge mamba python=3.10   || exit /B 
Collecting package metadata (current_repodata.json): ...working... done
Solving environment: ...working... done
## Package Plan ##
  environment location: C:\Miniconda38-x64
  added / updated specs:
    - mamba
    - python=3.10
The following packages will be downloaded:
    package                    |            build
    ---------------------------|-----------------
...
    libmamba-1.5.8             |       h3f09ed1_0         3.4 MB  conda-forge
    libmambapy-1.5.8           |  py310h04f2035_0         330 KB  conda-forge
    mamba-1.5.8                |  py310hd9d798f_0          67 KB  conda-forge
```

This is the first failed job by this problem. Note that there are some solving failed messages and mamba 0.24.0 was selected. (mamba is updated to 1.5.8 later):

https://ci.appveyor.com/project/ApacheSoftwareFoundation/arrow/builds/49867058#L88

```text
C:\projects\arrow>conda install -q -y -c conda-forge mamba python=3.10   || exit /B 
Collecting package metadata (current_repodata.json): ...working... done
Solving environment: ...working... failed with initial frozen solve. Retrying with flexible solve.
Solving environment: ...working... failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done
## Package Plan ##
  environment location: C:\Miniconda38-x64
  added / updated specs:
    - mamba
    - python=3.10
The following packages will be downloaded:
    package                    |            build
    ---------------------------|-----------------
...
    libmamba-0.24.0            |       h81a967f_1         2.9 MB  conda-forge
    libmambapy-0.24.0          |  py310hd80b381_1         527 KB  conda-forge
...
    mamba-0.24.0               |  py310h9376f3e_1          64 KB  conda-forge
```

I'm not sure whether this is related to this problem or not but our AppVeyor environment may be related. Because cleaning cache solved this problem.
